openapi/openapi_v2_backend_configuration_test.go

```diff
@@ -124,6 +124,302 @@ func TestGetHost(t *testing.T) {
 	})
 }
 
+func TestGetHostByRegion(t *testing.T) {
+	Convey("Given a specV2BackendConfiguration with a multi-region configuration", t, func() {
+		spec := &spec.Swagger{
+			SwaggerProps: spec.SwaggerProps{
+				Swagger: "2.0",
+			},
+			VendorExtensible: spec.VendorExtensible{
+				Extensions: spec.Extensions{
+					extTfProviderMultiRegionFQDN: "www.${region}.some-backend.com",
+					extTfProviderRegions:         "rst1",
+				},
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When getHostByRegion method is called with an existing region", func() {
+			host, err := specV2BackendConfiguration.getHostByRegion("rst1")
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+			Convey("And the host should be correct", func() {
+				So(host, ShouldEqual, "www.rst1.some-backend.com")
+			})
+		})
+		Convey("When getHostByRegion method is called with a NON existing region", func() {
+			_, err := specV2BackendConfiguration.getHostByRegion("nonExisting")
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldNotBeNil)
+			})
+			Convey("And the error should match the expected message", func() {
+				So(err.Error(), ShouldEqual, "region nonExisting not matching allowed ones [rst1]")
+			})
+		})
+		Convey("When getHostByRegion method is called with an empty region", func() {
+			_, err := specV2BackendConfiguration.getHostByRegion("")
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldNotBeNil)
+			})
+			Convey("And the error should match the expected message", func() {
+				So(err.Error(), ShouldEqual, "can't get host by region, missing region value")
+			})
+		})
+	})
+
+	Convey("Given a specV2BackendConfiguration with a NON multi-region configuration", t, func() {
+		spec := &spec.Swagger{
+			SwaggerProps: spec.SwaggerProps{
+				Swagger: "2.0",
+				Host:    "www.some-backend.com",
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When getHostByRegion method is called with a NON existing region", func() {
+			_, err := specV2BackendConfiguration.getHostByRegion("nonExisting")
+			Convey("Then the error returned should be nil", func() {
```

Comment:
should not be nil? doesn't match assertion

Reply:
done